pkg/apis/core/validation/validation.go

```diff
@@ -3263,19 +3264,26 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 // ValidateNodeSelectorRequirement tests that the specified NodeSelectorRequirement fields has valid data
 func ValidateNodeSelectorRequirement(rq core.NodeSelectorRequirement, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+	vals := rq.Values
 	switch rq.Operator {
 	case core.NodeSelectorOpIn, core.NodeSelectorOpNotIn:
-		if len(rq.Values) == 0 {
+		if len(vals) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("values"), "must be specified when `operator` is 'In' or 'NotIn'"))
 		}
 	case core.NodeSelectorOpExists, core.NodeSelectorOpDoesNotExist:
-		if len(rq.Values) > 0 {
+		if len(vals) > 0 {
 			allErrs = append(allErrs, field.Forbidden(fldPath.Child("values"), "may not be specified when `operator` is 'Exists' or 'DoesNotExist'"))
 		}
 
 	case core.NodeSelectorOpGt, core.NodeSelectorOpLt:
-		if len(rq.Values) != 1 {
+		if len(vals) != 1 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("values"), "must be specified single value when `operator` is 'Lt' or 'Gt'"))
+		} else {
+			for i := range vals {
+				if _, err := strconv.ParseInt(vals[i], 10, 64); err != nil {
+					allErrs = append(allErrs, field.Required(fldPath.Child("values"), "must be specified an integer value when `operator` is 'Lt' or 'Gt'"))
+				}
+			}
```

Comment:
/hold

standard questions when we want to tighten validation:
* does it prevent a no-op update of existing persisted objects that would fail this new validation? (would block storage migration tasks)
* does it prevent deletion of existing persisted objects with finalizers that would fail this new validation? (update is required to remove finalizers to allow deletion)
* does it prevent controllers from updating status of existing persisted objects that would fail this new validation?

we could consider ratcheting validation:
* apply this tightened validation on create
* apply this tightened validation on update where the old object passed this validation (don't let a valid object become invalid)
* skip this tightened validation on update where the old object fails this validation

c.f. https://github.com/kubernetes/kubernetes/issues/59984